### PR TITLE
Feature/27 & 28 admin merch disable enable

### LINKED ISSUE #33: feature/27-admin-merch-disable-enable
27. Admin Merchant Enable/Disable

As an admin,
When I visit the admin merchants index (/admin/merchants)
Then next to each merchant name I see a button to disable or enable that merchant.
When I click this button
Then I am redirected back to the admin merchants index
And I see that the merchant's status has changed

### PR DESCRIPTION
**Pull request by aplasket**
## Checklist before requesting a review
- [x] Verified Render deploy is functional
- [x] Pulled/merged main 
- [x] Tests for the changes have been added into spec files
- [x] Branch name follows our guidelines 
- [x] All tests running clean
- [x] Ran this in Rails S  - features are working properly
- [x] Test coverage is at 100% for models and features

## What changes or new behavior does this PR introduce (Bug fix, feature, update, etc)?
  - add enable/disable function to admin::merchants index
  - updated form_with to button_to 
  - added enabled/disabled sections (US28 completed too!!!)
  - updated factory bot/faker data for merchant names to be more fun!
  - update default merchant status to 1/disabled
 
## Bugs, Issues, etc:
 - [ ] check if there are known bugs or issue

## Other helpful info:

### CODE REVIEW

File: app/controllers/admin/merchants_controller.rb [hunk 1]
```diff
@@ -13,16 +13,22 @@ def edit
...
   def update
     merchant = Merchant.find(params[:id])
-    if params[:merchant][:name]
-      if merchant.update(merchant_params)
-        redirect_to admin_merchant_path(merchant)
-        flash[:success] = "Merchant has been successfully updated"
-      end
+    if params[:status] == "1"
+      merchant.update(status: 1)
+      redirect_back fallback_location: admin_merchant_path
```
Comment (AlexKiselich) [inline app/controllers/admin/merchants_controller.rb:18]: I have not seen this fallback location before, had to look that up, pretty cool and smart

File: db/schema.rb [hunk 2]
```diff
@@ -55,6 +55,7 @@
     t.string "name"
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
+    t.integer "status", default: 0
```
Comment (AlexKiselich) [inline db/schema.rb]: Should have added this in the beginning haha, my bad. Probably would have helped everyone

File: app/views/admin/merchants/index.html.erb [hunk 3]
```diff
@@ -1,3 +1,19 @@
-<% @merchants.each do |merchant| %>
-  <p><%= link_to merchant.name, admin_merchant_path(merchant) %></p>
-<% end %>
\ No newline at end of file
+<div id="enabled-merchants">
+  <h2>Enabled Merchants</h2>
+  <% @merchants.all_enabled.each do |merchant| %>
+    <div class="merchant-<%= merchant.id %>">
+      <p><%= link_to merchant.name, admin_merchant_path(merchant) %></p>
+      <%= button_to "Disable", admin_merchant_path(id: "#{merchant.id}", status: "1"), method: :patch %>
```
Comment (AlexKiselich) [inline app/views/admin/merchants/index.html.erb:6]: So if I am looking at this correctly you are passing in the params status back to the controller then you can use that in the if statement? I really like this and will look into button_to more!

**Review by AlexKiselich — APPROVED** [review]
Great work, that looked really hard haha

File: app/views/admin/merchants/index.html.erb (hunk 3 above)
Comment (aplasket) [reply app/views/admin/merchants/index.html.erb:6]: yup, exactly! it's passing in a param (status) with a value of "1" (disabled) and which is then used in the update method in the admin::merchant controller to update the status! pretty neat!

File: spec/features/admin/merchants/index_spec.rb
Comment (AlexKiselich) [inline spec/features/admin/merchants/index_spec.rb:1]: 2 for 1! Noice!

File: app/views/admin/merchants/index.html.erb (hunk 3 above)
Comment (kbergstrom78) [reply app/views/admin/merchants/index.html.erb:6]: button_to!!!!

File: db/migrate/20230604213602_add_status_to_merchants.rb
Comment (kbergstrom78) [inline db/migrate/20230604213602_add_status_to_merchants.rb:1]: I'm getting deja vu looking at this PR haha

**Review by kbergstrom78 — APPROVED** [review]

**Pull request merged**